src/librustc_codegen_ssa/mir/constant.rs

```diff
@@ -55,7 +56,15 @@ impl<'a, 'tcx, Bx: BuilderMethods<'a, 'tcx>> FunctionCx<'a, 'tcx, Bx> {
                         err
                     })
             }
-            _ => Ok(self.monomorphize(&constant.literal)),
+            ty::ConstKind::Value(value) => Ok(value),
+            _ => {
+                let const_ = self.monomorphize(&constant.literal);
+                if let ty::ConstKind::Value(value) = const_.val {
+                    Ok(value)
+                } else {
+                    bug!("encountered bad ConstKind in codegen");
```

Comment:
Please debug print the constant here so we know whats going on if we hit the ICE

Reply:
Also if at all possible, prefer `span_bug!` to `bug!`.